connector/dialect.tdd

```diff
@@ -1,376 +1,147 @@
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 &lt; %2 THEN %1
-	ELSE %2 END</formula>
-      <argument type='real' />
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='MIN' return-type='int'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='int' />
-      <argument type='int' />
-    </function>
-    <function group='numeric' name='PI' return-type='real'>
-      <formula>PI()</formula>
-    </function>
-    <function group='numeric' name='POWER' return-type='real'>
-      <formula><![CDATA[(CASE WHEN %1 < 0 AND FLOOR(%2) <> %2 THEN NULL ELSE POWER(%1,%2) END)]]></formula>
-      <argument type='real' />
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='RADIANS' return-type='real'>
-      <formula>RADIANS(%1)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='ROUND' return-type='real'>
-      <formula>ROUND(%1)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='ROUND' return-type='real'>
-      <formula>ROUND(CAST(%1 AS NUMERIC),CAST(%2 AS INTEGER))</formula>
-      <argument type='real' />
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='SIGN' return-type='int'>
-      <formula>CAST(SIGN(%1) AS SMALLINT)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='SIN' return-type='real'>
-      <formula>SIN(%1)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='SQRT' return-type='real'>
-      <formula><![CDATA[(CASE WHEN %1 < 0 THEN NULL ELSE SQRT(%1) END)]]></formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='SQUARE' return-type='real'>
-      <formula>(%1^2)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric' name='SQUARE' return-type='int'>
-      <formula>(%1^2)</formula>
+    <function group='aggregate' name='AVG' return-type='real'>
+      <formula>AVG(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='int' />
     </function>
-    <function group='numeric' name='TAN' return-type='real'>
-      <formula>TAN(%1)</formula>
-      <argument type='real' />
-    </function>
-    <function group='numeric;logical' name='ZN' return-type='real'>
-      <formula>COALESCE(%1, 0)</formula>
+    <function group='aggregate' name='AVG' return-type='real'>
+      <formula>AVG(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='real' />
     </function>
-    <function group='numeric;logical' name='ZN' return-type='int'>
-      <formula>COALESCE(%1, 0)</formula>
+    <function group='aggregate' name='COUNT' return-type='int'>
+      <formula>COUNT(%1)</formula>
+      <unagg-formula>(CASE WHEN %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
```

Comment:
Is this `<unagg-formula>` required? How is it used by Tableau in practice? The docs don't make it very clear. Is this being used in a query when `COUNT(x)` is present?

Reply:
After more testing, I found that the `<unagg-formula>` affects the GROUP BY clause in that if it's not set then the GROUP BY will have the aggregate function in it, which is why STDDEV was failing.  How it affects the GROUP BY clause when it IS set is not clear as so far it hasn't changed based on what I have in the unagg-formula.  

The description of the unagg-formula it `should represent a reasonable way of expressing the aggregate of a single value` which is correct in this case since count(NULL) is 0 and COUNT(N) is 1.